cfgmgr/buffermgr.cpp

```diff
@@ -152,47 +153,61 @@ task_process_status BufferMgr::doSpeedUpdateTask(string port, bool admin_up)
         return task_process_status::task_success;
     }
 
-    speed = m_speedLookup[port];
+    if (m_portPfcStatus.count(port) == 0)
+    {
+        // PORT_QOS_MAP is not ready yet. retry is required
+        SWSS_LOG_INFO("pfc_enable status is not available for port %s", port.c_str());
```

Comment:
It is possible that `pfc_enable` will not be configured but `speed` is configured. In that case, the notification will remain in `m_toSync` forever.
Maybe we can return `task_success` and call `doSpeedUpdateTask` from `doPortQosTableTask` each time `pfc_enable` is updated.

Reply:
Yes, that's a possible path. But even `pfc_enable` is not available when `speed` is configured, we will return `retry` here, and the main loop will skip this notification. When `pfc_enable` is configured eventually, the `speed` update notification will be consumed, right?
If we never have `pfc_enable` configured, yes, then the `BUFFER_PG` has no chance to be configured. But I think it also makes sense as we don't know on which PG to apply lossless profile.

Reply:
In case there is no `pfc_enable` configured, the notification will remain in `m_toSync`, and each time `doTask` is called the notification has to be checked. 
In addition, `doTask` will return once `doSpeedUpdateTask` returns a non-success value, leaving all the following notifications unhandled. Check [L506](https://github.com/Azure/sonic-swss/pull/2171/files#diff-ed9e5ec23caf0de71754703dac87fcf2be40f32e2443fa9e5617c1910665d7f2R505)
It's possible in your scenario if there are some inactive ports on the device because no `pfc_enable` configured on inactive ports.

Reply:
Oh, I see. The main loop is break here
https://github.com/Azure/sonic-swss/blob/841f00389b338e91ddc4de460ace4ff96adfa796/cfgmgr/buffermgr.cpp#L444-L447
Thanks for pointing out. I'll post an update soon.

Reply:
I removed the break if `task_status != task_process_status::task_success`. We should try to proceed the next port if one failed. 
With this change, we can consume other ports updates if one is lack of `pfc_enable`.

Reply:
Thanks.
A concern is that the notification will remain in `m_toSync` forever if there is no `pfc_enable` configured. It's true for inactive ports that are not connected to ToR switches. As a result, those notifications will be checked and skipped every time a new notification is received.
A solution can be to remove such notification from `m_toSync` and to call `doSpeedUpdateTask` from `doPortQosTableTask`

Reply:
Make sense. Fixed. Thanks